packages/client/lib/sync/fullsync.ts

```diff
@@ -1,34 +1,105 @@
 import { BN } from 'ethereumjs-util'
-import VM from '@ethereumjs/vm'
 import { Peer } from '../net/peer/peer'
 import { short } from '../util'
 import { Synchronizer, SynchronizerOptions } from './sync'
 import { BlockFetcher } from './fetcher'
 import { Block } from '@ethereumjs/block'
+import VM from '@ethereumjs/vm'
+import { DefaultStateManager } from '@ethereumjs/vm/dist/state'
+import { SecureTrie as Trie } from '@ethereumjs/trie'
 
 /**
  * Implements an ethereum full sync synchronizer
  * @memberof module:sync
  */
 export class FullSynchronizer extends Synchronizer {
+  private blockFetcher: BlockFetcher | null
+
   public vm: VM
+  public runningBlocks: boolean
 
-  private blockFetcher: BlockFetcher | null
+  private stopSyncing: boolean
+  private vmPromise?: Promise<void>
+
+  // Tracking vars for log msg condensation on zero tx blocks
+  private NUM_ZERO_TXS_PER_LOG_MSG = 50
+  public zeroTxsBlockLogMsgCounter: number = 0
```

Comment:
nit: I don't think need the `number` type annotation since it's being initialized with `0`